src-docs/src/views/header/header_sections.js

```diff
@@ -28,7 +28,7 @@ export default () => {
   const renderLogo = (
     <OuiHeaderLogo
       iconType="logoOpenSearch"
-      href="#"
+      href="https://oui.opensearch.org/latest/"
```

Comment:
Hover over the logo

<img width="600" alt="Screenshot 2023-04-18 at 09 19 09" src="https://user-images.githubusercontent.com/17855243/232662292-04c0ae03-f65b-43ff-ba0d-b1712a1351bf.png">